includes/amp-helper-functions.php

```diff
@@ -417,9 +418,16 @@ function amp_is_available() {
 	$warn = static function () {
 		static $already_warned_sources = [];
 
-		$likely_culprit_detector = Services::get( 'dev_tools.likely_culprit_detector' );
+		try {
+			$likely_culprit_detector = Services::get( 'dev_tools.likely_culprit_detector' );
```

Comment:
If this throws an exception, would it be better to register this one service?

Reply:
I suppose not because at present `\AmpProject\AmpWP\Infrastructure\ServiceBasedPlugin::register_services()` would bail if any services are registered.

Reply:
I think it would be preferable to have one single execution flow that we support and guarantee and let people know if they want to go against that, instead of creating multiple different ways of bootstrapping the services, leading to contextual bugs that are hard to pinpoint.